src/editmap.cpp

```diff
@@ -1831,6 +1831,170 @@ int editmap::mapgen_preview( real_coords &tc, uimenu &gmenu )
+            submap *srcsm = tmpmap.get_submap_at_grid( x, y, target.z );
+            destsm->is_uniform = false;
+            srcsm->is_uniform = false;
+
+            if( !destsm->vehicles.empty() ){
+                popup( _("Engine cannot support merging vehicles from two overmaps, please remove them from the OM tile.") );
+                return false;
+            }
+
+            destsm->delete_vehicles();
+            for( size_t i = 0; i < srcsm->vehicles.size(); i++ ) { // copy vehicles to real map
+                vehicle *veh1 = srcsm->vehicles[i];
+                veh1->smx = target_sub.x + x;
+                veh1->smy = target_sub.y + y;
+                veh1->smz = target.z;
+                destsm->vehicles.push_back( veh1 );
+                g->m.update_vehicle_cache( veh1, target.z );
+            }
+            srcsm->vehicles.clear();
+            g->m.update_vehicle_list( destsm, target.z );
+
+            int spawns_todo = 0;
+            for( size_t i = 0; i < srcsm->spawns.size(); i++ ) { // copy spawns
+                destsm->spawns.push_back( srcsm->spawns[i] );
+                spawns_todo++;
+            }
+
+            destsm->field_count = srcsm->field_count; // and count
+            std::memcpy( destsm->trp, srcsm->trp, sizeof( srcsm->trp ) ); // traps
+            std::memcpy( destsm->rad, srcsm->rad, sizeof( srcsm->rad ) ); // radiation
+            std::memcpy( destsm->lum, srcsm->lum, sizeof( srcsm->lum ) ); // emissive items
+
+
+            for( int sx = 0; sx < SEEX; ++sx ) {
+                for( int sy = 0; sy < SEEY; ++sy ) {
+                    for( auto &elem : srcsm->itm[sx][sy] ) {
+                        destsm->itm[sx][sy].push_back(elem);
+                    }
+                    //Don't cover existing crops, for farm upgrades
+                    //Don't destroy terrain to place grass/dirt
+                    if( change_sensitive && destsm->frn[sx][sy] != furn_str_id( "f_plant_seed" ) &&
+                        destsm->frn[sx][sy] != furn_str_id( "f_plant_seedling" ) &&
+                        destsm->frn[sx][sy] != furn_str_id( "f_plant_mature" ) &&
+                        destsm->frn[sx][sy] != furn_str_id( "f_plant_harvest" ) &&
+                        srcsm->ter[sx][sy] != ter_str_id( "t_grass" ) &&
+                        srcsm->ter[sx][sy] != ter_str_id( "t_dirt" ) ) {
+                               destsm->ter[sx][sy] = srcsm->ter[sx][sy];
+                               destsm->frn[sx][sy] = srcsm->frn[sx][sy];
+                       }
+                    destsm->fld[sx][sy] = srcsm->fld[sx][sy];
+                    destsm->cosmetics[sx][sy].swap( srcsm->cosmetics[sx][sy] );
+                }
+            }
+
+            // various misc variables
+            destsm->active_items = srcsm->active_items;
+
+            destsm->temperature = srcsm->temperature;
+            destsm->last_touched = calendar::turn;
+            destsm->comp = std::move( srcsm->comp );
```

Comment:
Are you sure `srcsm->comp` will not be used in any way afterwards?

Reply:
I'm not sure what you mean.  It's the computers listed in the generated submap to be moved to the real submap... not that we have any yet, but why would they be used afterwards in the temporary submap?  When the temp tinymap gets deleted would it clear any computers that were added to the final map?

Reply:
I'm just wondering why you copy everything else, but `move` this one field.